public/js/reducers/sources.js

```diff
@@ -100,10 +96,19 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       }
 
       return _updateText(state, action, [action.originalSource]);
+
     case "NAVIGATE":
-      // Reset the entire state to just the initial state, a blank state
-      // if you will.
-      return State();
+      if (!action.data) {
+        return state;
+      }
+      if (!state.url) {
+        return State()
+          .set("url", action.data.url);
+      }
+      if (state.url === action.data.url) {
+        return state;
+      }
+      return State().set("url", action.data.url);
```

Comment:
this seems like a lot of code to support the case where `will-navigate` is fired from the other events.

Also, i don't think that `url` should be set in `sources`. It probably makes sense in a debuggee reducer as at somepoint we might be debugging a worker URL as well and perhaps other things so there would be different debuggee data.